File: dehaze-python/app/models/entity/sys_ai_mcp_namespace.py
from typing import Any

from sqlalchemy import BigInteger, String
from sqlalchemy.dialects.mysql import JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel


class SysAiMcpNamespace(BaseModel):
    __tablename__ = "sys_ai_mcp_namespace"
    __table_args__ = {"comment": "外部MCP Server命名空间配置表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    server_id: Mapped[int] = mapped_column(
        BigInteger, nullable=False, comment="关联Server ID(关联sys_ai_mcp_server.id)"
    )
    namespace: Mapped[str] = mapped_column(
        String(128), nullable=False, comment="命名空间标识(工具分组,对齐McpNamespaceVO.name)"
    )
    tool_names: Mapped[Any | None] = mapped_column(
        JSON, nullable=True, comment="分组内工具名数组(JSON)"
    )

File: dehaze-go/internal/model/sys_ai_mcp_namespace.go
package model

import "encoding/json"

// SysAiMcpNamespace MCP Server 命名空间（工具分组）配置
type SysAiMcpNamespace struct {
	BaseModel
	ServerID  int64           `gorm:"column:server_id;type:bigint;not null;index:idx_server_id;comment:关联Server ID" json:"serverId"`
	Namespace string          `gorm:"column:namespace;type:varchar(128);not null;comment:命名空间标识" json:"namespace"`
	ToolNames json.RawMessage `gorm:"column:tool_names;type:json;comment:分组内工具名数组(JSON)" json:"toolNames"`
}

func (SysAiMcpNamespace) TableName() string {
	return "sys_ai_mcp_namespace"
}
